### PR TITLE
Refactor request forwarding and improve propagation of validation errors

### PR DESCRIPTION
**Pull request by bgeuken**
* DRY existing request tests
* Move controller code for forwarding requests to model
* Include validation error messages of bs request actions to bs request validation messages
* Add tests for request forwarding and propagating validation errors
* Refactor request related code

### CODE REVIEW

**Comment by bgeuken** [comment]
[P1: Validation failed: Bs request actions is invalid](https://trello.com/c/iv5UwCCx/715-p1-validation-failed-bs-request-actions-is-invalid)

File: src/api/app/models/bs_request.rb [hunk 1]
```diff
@@ -88,6 +88,7 @@ class SaveError < APIException
   validate :check_creator, on: [:create, :save!]
   validates :comment, length: { maximum: 65_535 }
   validates :description, length: { maximum: 65_535 }
+  validates_associated :bs_request_actions, message: ->(_, record) { record[:value].map { |r| r.errors.full_messages }.flatten.to_sentence }
```
Comment (ChrisBr) [inline src/api/app/models/bs_request.rb]: :+1: nice

File: src/api/app/models/bs_request.rb [hunk 2]
```diff
@@ -1169,10 +1169,10 @@ def forward_to(project:, package: nil, options: {})
                  source_package: action.target_package,
                  source_rev:     rev,
                  target_project: project,
-                 target_package: package }
+                 target_package: package,
+                 type:           'submit' }
```
Comment (ChrisBr) [inline src/api/app/models/bs_request.rb]: Why is the type now submit?

Comment (bgeuken) [reply src/api/app/models/bs_request.rb]: This was [previously](https://github.com/openSUSE/open-build-service/pull/4872/files#diff-a1bb0bdb9ac85a32ba6d96c4289745fcL428) done via
```ruby
  action = BsRequestActionSubmit.new(opts)
```

**Pull request merged**